src/lib.rs

```diff
@@ -189,343 +172,333 @@ where
         .map(|(x, (y, z))| (x, y, z))
 }
 
-macro_rules! impl_r2c {
-    ($ft:ty) => {
-        impl RealToComplex<$ft> {
-            /// Create a new RealToComplex FFT for input data of a given length. Returns an error if the length is not even.
-            pub fn new(length: usize) -> Res<Self> {
-                if length % 2 > 0 {
-                    let buffer_out = vec![Complex::zero(); length];
-                    let twiddles = Vec::new();
-                    let mut fft_planner = FftPlanner::<$ft>::new();
-                    let fft = fft_planner.plan_fft_forward(length);
-                    let scratch = vec![Complex::zero(); fft.get_inplace_scratch_len()];
-                    Ok(RealToComplex {
-                        twiddles,
-                        length,
-                        fft,
-                        buffer_out,
-                        scratch,
-                        is_even: false,
-                    })
-                }
-                else {
-                    let buffer_out = vec![Complex::zero(); length / 2 + 1];
-                    let twiddle_count = if length % 4 == 0 {
-                        length / 4
-                    } else {
-                        length / 4 + 1
-                    };
-                    let twiddles: Vec<Complex<$ft>> = (1..twiddle_count)
-                        .map(|i| compute_twiddle(i, length) * 0.5)
-                        .collect();
-                    let mut fft_planner = FftPlanner::<$ft>::new();
-                    let fft = fft_planner.plan_fft_forward(length / 2);
-                    let scratch = vec![Complex::zero(); fft.get_outofplace_scratch_len()];
-                    Ok(RealToComplex {
-                        twiddles,
-                        length,
-                        fft,
-                        buffer_out,
-                        scratch,
-                        is_even: true,
-                    })
-                }
-            }
+impl<T: FftNum>  RealToComplex<T> {
+    /// Create a new RealToComplex FFT for input data of a given length. Returns an error if the length is not even.
```

Comment:
This comment isn't accurate anymore, since it always returns Ok()

Reply:
That looks as it's not from the latest version, this part changed a lot in the next couple of commits.

Reply:
And thanks for looking!